lib/_http_outgoing.js

```diff
@@ -109,6 +109,24 @@ function OutgoingMessage() {
 Object.setPrototypeOf(OutgoingMessage.prototype, Stream.prototype);
 Object.setPrototypeOf(OutgoingMessage, Stream);
 
+Object.defineProperty(OutgoingMessage.prototype, 'writableFinished', {
+  /*
+    !socket is true if msg has not connected or has closed
+    finished checks that end() has been called
+    outputSize === 0 checks that there is no pending data
+
+    The logic here is a bit complicated.
+
+    We can't just rely on !socket since it will be a false positive if
+    still not connected yet.
+    However, if finished is true that means that some data must have been
+    queued on the msg object which means that outputSize > 0 unless a socket
+    has been assigned (even if it no longer is assigned).
```

Comment:
As you say yourself, this is a bit complicated, so I think it would be a good idea for unit tests to come along with this (ideally for all of the conditions below)

Reply:
That's over my head unfortunately. I've made the same suggestion here https://github.com/jshttp/on-finished/pull/31... maybe that will provide unit tests... don't know

Reply:
I've improved the description a bit

Reply:
tests added